### PR TITLE
Logout-URI Compatibility for Keycloak >= 18.0

### PR DESCRIPTION
**Pull request by serumk**

### CODE REVIEW

**Comment by MArK1done** [comment]
This actually good work, had to make the changes myself in a project where I'm using keycloak 19.0. Later came here to make a pull request only to find it has been done an hour before me :).

**Comment by mariovalney** [comment]
Hi @serumk .
Thanks for your contribution.

Would you mind to check my comments?

**Comment by serumk** [comment]
Hi Mario,


> Am 22.09.2022 um 18:30 schrieb Mário Valney ***@***.***>:
> 
> Hi @serumk <https://github.com/serumk> .
> Thanks for your contribution.
> 
It's a pleasure to give something back!
> Would you mind to check my comments?
> 
Sorry, which comments? Am I blind, because I can't see any comments..

best, Rudi

**Comment by mariovalney** [comment]
> Sorry, which comments? Am I blind, because I can't see any comments.. best, Rudi

On commits / code review:
https://docs.github.com/pt/pull-requests/collaborating-with-pull-requests/reviewing-changes-in-pull-requests/reviewing-proposed-changes-in-a-pull-request

**Comment by serumk** [comment]
Did you submit your review? I followed the instructions, but still see no comments, just my changes.

File: src/Controllers/AuthController.php [hunk 1]
```diff
@@ -30,9 +30,8 @@ public function login()
      */
     public function logout()
     {
-        KeycloakWeb::forgetToken();
-
         $url = KeycloakWeb::getLogoutUrl();
+        KeycloakWeb::forgetToken();
```
[the remark this answers is not in the document]
Comment (serumk) [reply src/Controllers/AuthController.php:34]: Unfortunately yes. For the new logout-URL we need the id-token (see https://www.keycloak.org/2022/04/keycloak-1800-released), so I have to construct the URL before forgetting the tokens.

File: src/Services/KeycloakService.php [hunk 2]
```diff
@@ -172,9 +172,13 @@ public function getLogoutUrl()
         }
 
         $params = [
-            'client_id' => $this->getClientId(),
-            'redirect_uri' => $this->redirectLogout,
+            'client_id' => $this->getClientId()
         ];
+        $idToken= session()->get(self::KEYCLOAK_SESSION . ".id_token");
```
[the remark this answers is not in the document]
Comment (serumk) [reply src/Services/KeycloakService.php:177]: Ah, ok - I missed this function.

**Pull request merged**